azurerm/resource_arm_container_group.go

```diff
@@ -0,0 +1,317 @@
+package azurerm
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/arm/containerinstance"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceArmContainerGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmContainerGroupCreate,
+		Read:   resourceArmContainerGroupRead,
+		Update: resourceArmContainerGroupCreate,
+		Delete: resourceArmContainerGroupDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"ip_address_type": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+				ValidateFunc: validation.StringInSlice([]string{
+					"Public",
+				}, true),
+				Default: "Public",
+			},
+
+			"os_type": {
+				Type:             schema.TypeString,
+				Required:         true,
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+				ValidateFunc: validation.StringInSlice([]string{
+					"windows",
+					"linux",
+				}, true),
+			},
+
+			"tags": tagsSchema(),
+
+			"ip_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"container": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"image": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"cpu": {
+							Type:     schema.TypeFloat,
+							Required: true,
+						},
+
+						"memory": {
+							Type:     schema.TypeFloat,
+							Required: true,
+						},
+
+						"port": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							ValidateFunc: validation.IntBetween(1, 65535),
+						},
+
+						"protocol": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"tcp",
+								"udp",
+							}, true),
```

Comment:
given the validation function allows for differences in case (the `true`) can we add a `DiffSuppressFunc` to this to ignore the case?

Reply:
Done.